docs/architecture/ADR-007-bad-debt.md

```diff
@@ -0,0 +1,92 @@
+# ADR 005: Liquidation
+
+## Changelog
+
+- December 10, 2021: Initial Draft (@toteki)
+
+## Status
+
+Proposed
+
+## Context
+
+Debt positions on Umee always start out over-collateralized, but if sufficient interest accrues or asset prices fluctuate too quickly, some borrowers may reach a state where the total value of their collateral is less than the value of their borrowed assets.
+
+Such debt (now under-collateralized) may become over-collateralized on its own if asset prices rebound in the right direction, but it is also eligible for liquidation in its current state.
+When fully liquidated, accounts whose `BorrowedValue` is greater than `CollateralValue` (adjusted for liquidation incentive) will result in an account with nonzero total borrows but zero collateral, thus no incentive for repayment.
+
+It is in the interests of the overall system to repay such _bad debt_ using reserves, in order to prevent it accruing interest forever and damaging the health of the lending pool.
+
+## Alternatives
+
+### Repayment Eligibility Condition
+
+One consideration to make is whether borrowers should be eligible for bad debt repayment whenever `Collateral Value < Borrow Value`, or if eligibility should be delayed until `Collateral Value == 0`.
+
+In the former case, the system would also confiscate any remaining collateral, as though it were a liquidator, and would have to process the collateral in a way of our choosing (e.g. exchange for base assets, and add them to reserves).
+
+In the latter case, liquidators would first have liquidate the portion of the debt that can be exchanged for collateral, reducing the total burden on reserves when the borrow is being repaid.
+
+Also in the latter case, if liquidators do not (for whatever reason) liquidate until the under-collateralized account's collateral is zero, its borrows would continue to accrue interest and remain ineligible for bad debt repayment.
+This should not be a common case, as long as liquidation incentives exist, but might happen if liquidators are sparse, or if the remaining collateral is so small that it is not worth the gas fee to retrieve.
+
+### Checking for Repayment Eligibility
+
+A second consideration is when to check for borrowers eligibility for bad debt repayment. The two likely options are immediately during `LiquidateBorrow`, or during `EndBlock` every `InterestEpoch`. A third option might be to check periodically, but on a separately controlled interval `BadDebtRepayEpoch`.
+
+The advantage of an immediate check during `LiquidateBorrow` is that only the borrow being liquidated needs to be checked for eligibility, instead of periodically iterating over all borrows. Additioanlly, `CollateralValue` has already been calculated in that function.
```

Comment:
```suggestion
The advantage of an immediate check during `LiquidateBorrow` is that only the borrow being liquidated needs to be checked for eligibility, instead of periodically iterating over all borrows. Additionally, `CollateralValue` has already been calculated in that function.
```